src/main/java/net/snowflake/client/jdbc/SFAsyncResultSet.java

```diff
@@ -357,6 +357,9 @@ public boolean isWrapperFor(Class<?> iface) throws SQLException {
   public List<SnowflakeResultSetSerializable> getResultSetSerializables(long maxSizeInBytes)
       throws SQLException {
     raiseSQLExceptionIfResultSetIsClosed();
-    throw new SnowflakeLoggedFeatureNotSupportedException(session);
+    getRealResults();
```

Comment:
Wait the async query done and call `getResultSetSerializables()` for the real ResultSet.